common/src/BasePlugin.php

```diff
@@ -103,19 +103,60 @@ public function isWoocommerceActivated(): bool {
         return class_exists('woocommerce');
     }
 
+    public function getActiveGtinPlugin() {
+        return GtinHandler::getActivePlugin();
+    }
+
     public function getProductMetaKeys(): array {
         global $wpdb;
-        return $wpdb->get_col("
+        $meta_keys = $wpdb->get_col("
             SELECT DISTINCT(pm.meta_key)
             FROM {$wpdb->posts} p
             LEFT JOIN {$wpdb->postmeta} pm ON p.ID = pm.post_id
             WHERE
                 p.post_type = 'product'
                 AND pm.meta_key <> ''
         ");
+        return array_map(
+            function ($value) {
+                return [
+                    'type' => 'meta',
+                    'name' => $value
+                ];
+            },
+            $meta_keys
+        );
     }
 
-    public function hasActiveGtinPlugin(): bool {
-        return GtinHandler::hasActivePlugin();
+    public function getProductKeys() {
+        return array_merge($this->getProductMetaKeys(), $this->getCustomAttributes());
+    }
+
+    private function getCustomAttributes(): array {
+        global $wpdb;
+        $custom_attributes = [];
+        $sql = "
+            SELECT meta.meta_id, meta.meta_key as name, meta.meta_value as type 
+            FROM {$wpdb->postmeta} meta
+            JOIN {$wpdb->posts} posts
+            ON meta.post_id = posts.id 
+            WHERE posts.post_type LIKE '%product%' 
+            AND meta.meta_key='_product_attributes';";
+
+        $data = $wpdb->get_results($sql);
+        if (!empty($data)) {
+            foreach ($data as $value) {
+                $product_attr = unserialize($value->type);
+                if (!empty($product_attr)) {
```

Comment:
1. I would make this `if (is_array($product_attr))` so that if the meta value contains something that's not an array, you won't try to foreach over it.
1. Instead of nesting so deeply, I would recommend doing `if (!is_array(...)) { continue; }`.